test/jdk/java/awt/MouseInfo/ContainerResizeMousePositionTest.java

```diff
@@ -0,0 +1,138 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4009555
+ * @summary Unit test for a new method in Container class: getMousePosition(boolean)
+ *          while Container resized.
+ */
+
+public class ContainerResizeMousePositionTest {
+    private static Frame frame;
+    private static Button button;
+    private static Robot robot;
+    private static volatile Point frameLocation;
+    private static volatile Point newLoc;
+    private static boolean testSucceeded = false;
+
+    private static final CountDownLatch eventCaught = new CountDownLatch(1);
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> createAndShowUI());
+            robot.waitForIdle();
+            robot.delay(1000);
+            testUI();
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void createAndShowUI() {
+        frame = new Frame("Testing getMousePosition() after resize");
+        button = new Button("Button");
+        frame.setLayout(new BorderLayout());
+        frame.add(button);
+        frame.setSize(200, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void testUI() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frameLocation = frame.getLocationOnScreen();
+            newLoc = new Point(frame.getWidth() + 10, frame.getHeight() + 10);
+        });
+
+        robot.mouseMove(frameLocation.x + newLoc.x, frameLocation.y + newLoc.y);
+        EventQueue.invokeAndWait(() -> {
+            button.addComponentListener(new ResizeAdapter());
+            frame.setSize(frame.getWidth() * 2, frame.getHeight() * 2);
+            frame.validate();
+        });
+        robot.waitForIdle();
+        robot.delay(500);
+
+        if (!eventCaught.await(2, TimeUnit.SECONDS)) {
+            if (!testSucceeded) {
```

Comment:
These two conditions look independent.

If `eventCaught.await` returned `false`, the test should fail with a timeout.

Otherwise, the test fails if `testSucceeded` didn't become `true`.

Reply:
Did you mean this way ?

```
if (eventCaught.await(2, TimeUnit.SECONDS)) {
    if (!testSucceeded) {
        throw new RuntimeException("Test Failed: Container.getMousePosition(boolean)"
                                   + " returned incorrect result while Container resized");
    }
} else {
    throw new RuntimeException("Test failed: Didn't receive"
                               + " Component Resized Event");
}
```

Reply:
I'd rather not put them into one condition either, even though the condition above looks correct.

```java
        if (!eventCaught.await(2, TimeUnit.SECONDS)) {
            throw new RuntimeException("componentResized isn't received within a timeout");
        }

        if (!testSucceeded) {
            throw new RuntimeException("Container.getMousePosition(boolean)"
                                       + " returned incorrect result while Container resized");
        }
```

Reply:
After re-reading the comment it makes sense it keep it separate. Thanks. Updated.